docs/current_docs/cookbook/snippets/custom-spans/go/main.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"context"
+	"dagger/my-module/internal/dagger"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type MyModule struct{}
+
+// Create an alpine container, write three files, and emit custom spans for each
+func (m *MyModule) Foo(ctx context.Context) *dagger.Directory {
+	// define the files to be created and their contents
+	files := map[string]string{
+		"file1.txt": "foo",
+		"file2.txt": "bar",
+		"file3.txt": "baz",
+	}
+
+	// set up an alpine container with the directory mounted
+	container := dag.Container().
+		From("alpine:latest").
+		WithDirectory("/results", dag.Directory()).
+		WithWorkdir("/results")
+	for name, content := range files {
+		// create a span for each file creation operation
+		_, span := Tracer().Start(ctx, "create-file", trace.WithAttributes(attribute.String("file.name", name)))
+		defer span.End()
+		// create the file and add it to the container
+		container = container.WithNewFile(name, content)
```

Comment:
Ah this is a gotcha - the point of a span is to measure the time spent doing something, but in this case the duration will always be 0 because the `WithNewFile` is lazy. Same problem exists in the other examples.

Might even be worth documenting that gotcha considering how easily it can be misunderstood.

Spans are typically used to separate a bunch of suites that are running in parallel, with each branch waiting for completion. [Here's a good example.](https://github.com/dagger/dagger/blob/6e652c2e44c4b2cd46e7914e099d51fe30f90aba/.dagger/sdk_go.go#L22-L48) An example that does that would probably click more quickly as to when you might want to use spans.

Reply:
@vito thanks for the feedback. I've modified the example and added a warning, can you please review it again? It seems to be fine in Go, but I'm not able to get the Python and Typescript versions working so have asked separately for help with those.